src/main/java/org/blyznytsia/bpp/AutowiredAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,42 @@
+package org.blyznytsia.bpp;
+
+import lombok.extern.slf4j.Slf4j;
+import org.blyznytsia.annotation.Autowired;
+import org.blyznytsia.context.ApplicationContext;
+import org.blyznytsia.exception.BeanConfigurationException;
+
+/**
+ * Implementation of {@link BeanPostProcessor} interface that injects beans into fields annotated
+ * with {@link org.blyznytsia.annotation.Autowired}
+ */
+@Slf4j
+public class AutowiredAnnotationBeanPostProcessor implements BeanPostProcessor {
+
+  /**
+   * Injects beans into fields annotated with {@link org.blyznytsia.annotation.Autowired}
+   *
+   * @param bean bean to be configured
+   * @param context application context to fetch beans from for injections
+   */
+  @Override
+  public void configure(Object bean, ApplicationContext context) {
+    log.debug("Configuring bean of {} type", bean.getClass());
+    try {
+      for (var field : bean.getClass().getDeclaredFields()) {
+        if (field.isAnnotationPresent(Autowired.class)) {
+          field.setAccessible(true);
+          var value = context.getBean(field.getType());
```

Comment:
I thought that this postProcessor will be invoked before bean will be put in the context